src/library_wasmfs.js

```diff
@@ -86,7 +86,34 @@ var WasmfsLibrary = {
       // User code should not be using FS.cwd().
       // For file preloading, cwd should be '/' to begin with.
       return '/';
-    }
+    },
+
+#if FORCE_FILESYSTEM
+    // Full JS API support
+    mkdir: function(path, mode) {
+      // Adjust mode in the same way as the JS FS
+      mode = mode !== undefined ? mode : 511 /* 0777 */;
```

Comment:
It would be nice to have a comment saying what 511 is and why it is chosen, and maybe also to use `cDefine`s rather than the raw numbers. Also, what is the commented out 0777 for?

Reply:
Good questions - I actually don't know. This just matches/copies the JS code, which does not have any more insights.

I think it's good enough to migrate the old JS code to here more or less as is, without trying to make significant improvements in it. I'm not opposed to improvements, but there will be a lot of possible ones, and I'm not sure it's worth slowing things down while we struggle to reach parity.

`0777` is in octal. I think that is the "natural" way to represent the mode in some places, so it's written for documentation purposes. (But again, I'm not an expert on that code.)

Reply:
Separately I see the param is not used atm, so I'll remove it entirely.

Reply:
Do you still care about IE? Octal literals are now supported everywhere else: https://caniuse.com/mdn-javascript_grammar_octal_numeric_literals So you could change this to `0o777`.

Reply:
Or you can do `{{{ 0o777 }}}` to have to converted to 511 at compile time.

Reply:
I think we do still want to support IE, but it's been an easy decision to do so since it has had little cost (over the other legacy stuff we want to support). Perhaps we should reconsider that at some point.